docs/source/delayed-overview.rst

```diff
@@ -1,6 +1,86 @@
 Overview
 ========
 
+Motivation and Example
+----------------------
+
+Dask.delayed lets you parallelize custom code.  It is useful whenever your
+problem doesn't quite fit a high-level parallel object like dask.array or
+dask.dataframe but could still benefit from parallelism.  Dask.delayed works by
+delaying your function evaluations and putting them into a dask graph.
+Dask.delayed is useful when wrapping existing code or when handling
+non-standard problems.
+
+Consider the following example:
+
+.. code-block:: python
+
+    def inc(x):
+        return x + 1
+
+    def double(x):
+        return x + 2
+
+    def add(x, y):
+        return x + y
+
+    data = [1, 2, 3, 4, 5]
+
+    output = []
+    for x in data:
+        a = inc(x)
+        b = double(x)
+        c = add(a, b)
+        output.append(c)
+
+    total = sum(output)
+
+As written this code runs sequentially in a single thread.  However we see that
+a lot of this could be executed in parallel.  We use the ``delayed`` function
+to parallelize this code by turning it into a dask graph.  We slightly modify
+our code by wrapping functions in ``delayed``.  This delays the execution of
+the function and generates a dask graph instead.
+
+.. code-block:: python
+
+   from dask import delayed
```

Comment:
For this example, I would find it easier to add the `@delayed ` decorator to the functions. Faster to type also... If folks don't understand decorators, they may also get confused by the syntax you use here. The decorator seems easier to use even if it just seems like magic.